src/App.tsx

```diff
@@ -1,223 +1,174 @@
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  FC,
+  useMemo,
+  useCallback,
+} from 'react';
+import { getTodos, createTodo, deleteTodo } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [completedFilter, setCompletedFilter,
+  ] = useState<FilterType>(FilterType.ALL);
+  const [title, setTitle] = useState<string>('');
+  const [isAdding, setIsAdding] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
 
-export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => {
+          setErrorMessage('Unable to load todos');
+        });
+    }
+
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const onDeleteTodo = (todoId: number) => {
+    setDeletingTodoIds(prev => [...prev, todoId]);
+    deleteTodo(todoId)
+      .then(() => (
+        setTodos(allTodos => allTodos.filter(todo => todo.id !== todoId))
+      ))
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      })
+      .finally(() => (
+        setDeletingTodoIds(prev => prev.filter(id => id !== todoId))));
+  };
+
+  const onFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setErrorMessage('Title can\'t be empty');
+
+      return;
+    }
+
+    if (user) {
+      setTempTodo({
+        id: 0,
+        title,
+        completed: false,
+        userId: user.id,
+      });
+      setIsAdding(true);
+
+      createTodo({
+        title,
+        userId: user.id,
+        completed: false,
+      })
+        .then(newTodo => {
+          setTodos(prev => [...prev, newTodo]);
+        })
+        .catch(() => setErrorMessage('Unable to add a todo'))
+        .finally(() => {
+          setIsAdding(false);
+          setTitle('');
+          setTempTodo(null);
+        });
+    }
+  };
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (completedFilter) {
+        case FilterType.ALL:
+          return todo;
```

Comment:
You still filter the array if `completedFilter` equals `FilterType.ALL`. Don't do it